sdk/avalon_sdk/connector/blockchains/fabric/event_listener.py

```diff
@@ -124,3 +126,36 @@ async def stop_event_handling(self, seconds):
         await asyncio.sleep(seconds)
         self._channel_event_hub.unregisterChaincodeEvent(self._event_regid)
         self._channel_event_hub.disconnect()
+
+    async def get_single_event(self):
+        """
+        Start event listener and listen for particular event.
+        Once we got the particular event then unregister and
+        close the event listenerhub.
+        """
+        def _event_handler(event, block_num, txnid, status):
+            logger.debug(
+                'Event: {0}\nblock_num: {1}\ntxid: {2}\nstatus: {3}'.format(
+                    event, block_num, txnid, status))
+            try:
+                is_done = self._handler(event, block_num, txnid, status)
+                if is_done:
+                    self._channel_event_hub.unregisterChaincodeEvent(
+                        self._event_regid
+                    )
+                    self._channel_event_hub.disconnect()
+                self._last_block = block_num
+            except Exception as ex:
+                logger.error('Handler error: {0}'.format(ex))
+
+        self._event_regid = self._channel_event_hub.registerChaincodeEvent(
+            self._chaincode, self._event, start=self._last_block,
+            onEvent=_event_handler)
+        logger.info('Event handler registered!')
+        try:
+            await self._channel_event_hub.connect(False)
```

Comment:
When does this function gets unblocked. Is it when disconnect() is called from event_handler. Not very obvious. Could you please add a comment in code.

Reply:
connect is async call please do refer the fabric-sdk-python for information.

Reply:
ok. It was not obvious to me looking at the code how this function is waiting to get a single event. I couldn't find Fabric Python SDK doc. Instead I went through the documentation in this link https://hyperledger.github.io/fabric-sdk-node/release-1.4/ChannelEventHub.html 

According to this `registerChainCodeEvent()` should be called after calling `connect()`.  This is what `connect` documentation mentions -> "Register event listeners and the error callbacks by calling any one of the registerBlockEvent, registerTxEvent or registerChaincodeEvent methods, after calling connect()"

Reply:
That is fabric node sdk documentation. Refer the sample code here - https://github.com/hyperledger/fabric-sdk-py/blob/master/test/integration/e2e_test.py